src/Providers/Gemini/Handlers/Structured.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace EchoLabs\Prism\Providers\Gemini\Handlers;
+
+use EchoLabs\Prism\Exceptions\PrismException;
+use EchoLabs\Prism\Providers\Gemini\Maps\FinishReasonMap;
+use EchoLabs\Prism\Providers\Gemini\Maps\MessageMap;
+use EchoLabs\Prism\Structured\Request;
+use EchoLabs\Prism\ValueObjects\Messages\UserMessage;
+use EchoLabs\Prism\ValueObjects\ProviderResponse;
+use EchoLabs\Prism\ValueObjects\ResponseMeta;
+use EchoLabs\Prism\ValueObjects\Usage;
+use Illuminate\Http\Client\PendingRequest;
+use Illuminate\Http\Client\Response;
+use Throwable;
+
+class Structured
+{
+    public function __construct(protected PendingRequest $client) {}
+
+    public function handle(Request $request): ProviderResponse
+    {
+        try {
+            $request = $this->appendMessageForJsonMode($request);
+            $response = $this->sendRequest($request);
+            $this->validateResponse($response);
+
+            return $this->createResponse($response);
+        } catch (Throwable $e) {
+            throw PrismException::providerRequestError($request->model, $e);
+        }
+    }
+
+    public function sendRequest(Request $request): Response
+    {
+        $endpoint = "{$request->model}:generateContent";
+
+        $payload = (new MessageMap($request->messages, $request->systemPrompt))();
+
+        $generationConfig = array_filter([
+            'temperature' => $request->temperature,
+            'topP' => $request->topP,
+            'maxOutputTokens' => $request->maxTokens,
+            'response_mime_type' => 'application/json',
+        ]);
+
+        if ($generationConfig !== []) {
+            $payload['generationConfig'] = $generationConfig;
+        }
```

Comment:
Hey This extra check is not needed here as.
generation config will never be empty.

```php
$payload['generationConfig'] = array_filter([
            'temperature' => $request->temperature,
            'topP' => $request->topP,
            'maxOutputTokens' => $request->maxTokens,
            'response_mime_type' => 'application/json',
        ]);
  ```